### PR TITLE
Add some new tests

### PR DESCRIPTION
**Pull request by dufferzafar**
- [x] `http_request_translator.url`
- [x] `http_request_translator.plugin_manager`

~~`http_request_translator.translator` (currently working on this)~~

Refs #33

### CODE REVIEW

**Comment by dufferzafar** [comment]
To allow testing of the `translator.process_arguments` function, we'll have to restructure the code so that the `ArgumentParser` gets created in a function which can be imported separately. I don't think mixing code changes and tests in a same pull will be right, so I'll be doing that once these changes get merged!

**Comment by dhruvagarwal** [comment]
@dufferzafar This can be done even without changing the code. I'll post a prototype of it in a while.

File: tests/test_plugin_manager.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import unittest
+from textwrap import dedent
+
+from http_request_translator import (
```
Comment (DePierre) [inline tests/test_plugin_manager.py]: Can you inline these imports? It should fit under 120 characters.

File: tests/test_plugin_manager.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+    ###
+    # plugin_manager.generate_script
+    ###
+    def test_generate_script(self):
+        script = pmgr.generate_script(
+            "bash",
+            headers=["host:github.com"],
+            details=dict(path="file.txt", method="GET")
+        )
```
Comment (DePierre) [inline tests/test_plugin_manager.py]: Can you move that ')' at the end of the previous line?

File: tests/test_plugin_manager.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
+import unittest
+from textwrap import dedent
+
+from http_request_translator import (
+    plugin_manager as pmgr,
```
Comment (DePierre) [inline tests/test_plugin_manager.py]: Just curious to know why renaming the package is required?

**Comment by DePierre** [comment]
@dufferzafar Sorry for the delay. I have checked your PR and it looks good to me. I have added a couple of comments/questions if you can have a look.

I see that the unittests for url are incomplete but it's nice that you added some basic ones ;)

When you will have review the comments, I'll take a second look and merge it

**Comment by DePierre** [comment]
@dufferzafar Are you still available to finish your PR?

**Comment by DePierre** [comment]
@dufferzafar Thank you for your additional tests :) really appreciated and sorry for the delay...

**Pull request merged**